src/EditorFeatures/Core.Wpf/Utilities/RoslynServiceExtensions.cs

```diff
@@ -0,0 +1,186 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.Shell
+{
+    internal static class RoslynServiceExtensions
+    {
+        /// <inheritdoc cref="RoslynServiceExtensions.GetService{TService, TInterface}(IServiceProvider, JoinableTaskFactory, bool)"/>
+        public static TInterface GetServiceOnMainThread<TService, TInterface>(this IServiceProvider serviceProvider)
+        {
+            var service = serviceProvider.GetService(typeof(TService));
+            if (service is null)
+                throw new ServiceUnavailableException(typeof(TService));
+
+            if (service is not TInterface @interface)
+                throw new ServiceUnavailableException(typeof(TInterface));
+
+            return @interface;
+        }
+
+        /// <summary>
+        /// Returns the specified service type from the service.
+        /// </summary>
+        public static TServiceType GetServiceOnMainThread<TServiceType>(this IServiceProvider serviceProvider) where TServiceType : class
+            => serviceProvider.GetServiceOnMainThread<TServiceType, TServiceType>();
+
+        /// <summary>
+        /// Gets a service interface from a service provider.
+        /// </summary>
+        /// <typeparam name="TService">The service type</typeparam>
+        /// <typeparam name="TInterface">The interface type</typeparam>
+        /// <param name="serviceProvider">The service provider</param>
+        /// <returns>The requested service interface. Never <see langword="null"/>.</returns>
+        /// <exception cref="ServiceUnavailableException">
+        /// Either the service could not be acquired, or the service does not support
+        /// the requested interface.
+        /// </exception>
+        public static TInterface GetService<TService, TInterface>(
+            this IServiceProvider serviceProvider,
+            JoinableTaskFactory joinableTaskFactory)
```

Comment:
can we have some extensions that also take an IThreadingContext instead of hte JTF?

Reply:
I would prefer not. These are the extensions under consideration for future addition to the SDK.

Reply:
They can be in a different type :-). But I don't feel strongly. But it's more about the convenience for us.